src/rgw/rgw_lc.cc

```diff
@@ -828,15 +824,47 @@ int RGWLC::handle_multipart_expiration(rgw::sal::Bucket* target,
   params.ns = RGW_OBJ_NS_MULTIPART;
   params.access_list_filter = &mp_filter;
 
+  const auto event_type = rgw::notify::ObjectExpirationAbortMPU;
+  std::string version_id;
+
   auto pf = [&](RGWLC::LCWorker* wk, WorkQ* wq, WorkItem& wi) {
+    int ret{0};
     auto wt = boost::get<std::tuple<lc_op, rgw_bucket_dir_entry>>(wi);
     auto& [rule, obj] = wt;
     if (obj_has_expired(this, cct, obj.meta.mtime, rule.mp_expiration)) {
       rgw_obj_key key(obj.key);
       std::unique_ptr<rgw::sal::MultipartUpload> mpu = target->get_multipart_upload(key.name);
-      int ret = mpu->abort(this, cct, null_yield);
+      std::unique_ptr<rgw::sal::Object> sal_obj
+	= target->get_object(key);
+      std::unique_ptr<rgw::sal::Notification> notify
+	= driver->get_notification(
+	  this, sal_obj.get(), nullptr, event_type,
+	  target, lc_id,
+	  const_cast<std::string&>(target->get_tenant()),
+	  lc_req_id, null_yield);
+
+      ret = notify->publish_reserve(this, nullptr);
+      if (ret < 0) {
+				ldpp_dout(wk->get_lc(), 0)
+					<< "ERROR: reserving persistent notification for abort_multipart_upload, ret=" << ret
+					<< ", thread:" << wq->thr_name()
+					<< ", meta:" << obj.key
+					<< dendl;
+      }
+
+      ret = mpu->abort(this, cct, null_yield);
       if (ret == 0) {
-        if (perfcounter) {
+
+        ret = notify->publish_commit(
```

Comment:
if `publish_reserved()` failed, the commit could fail as well, we can avoid calling it here

Reply:
amended, as above